File: app/src/main/java/com/app/rush47/HomeActivity.java
package com.app.rush47;

import android.content.Intent;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.app.rush47.utils.UserLocalStore;

/**
 * Minimal placeholder Home screen so the auth flow has somewhere to land.
 * The real dashboard (tabs, Ludo, Lottery, Wallet etc.) is the next part
 * we build on top of this, following the module list already agreed on.
 */
public class HomeActivity extends AppCompatActivity {

    @Override
    protected void onCreate(android.os.Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);

        UserLocalStore userLocalStore = new UserLocalStore(this);
        TextView welcome = findViewById(R.id.home_welcome_text);
        String firstName = userLocalStore.getLoggedInUser().getFirstName();
        if (firstName != null && !firstName.isEmpty()) {
            welcome.setText(getString(R.string.home_welcome) + " " + firstName);
        }

        Button logoutButton = findViewById(R.id.logout_button);
        logoutButton.setOnClickListener(v -> {
            userLocalStore.clearUserData();
            startActivity(new Intent(getApplicationContext(), LoginActivity.class));
            finish();
        });
    }
}

File: app/src/main/java/com/rush47/tournament/ui/HomeActivity.kt
package com.rush47.tournament.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.rush47.tournament.databinding.ActivityHomeBinding

class HomeActivity : AppCompatActivity() {

    private lateinit var binding: ActivityHomeBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityHomeBinding.inflate(layoutInflater)
        setContentView(binding.root)

        val prefs = getSharedPreferences("rush47_prefs", MODE_PRIVATE)
        val firstName = prefs.getString("first_name", null)

        binding.tvWelcome.text = if (!firstName.isNullOrEmpty()) {
            "Welcome, $firstName! 👋"
        } else {
            "Welcome to RUSH 47! 👋"
        }

        binding.btnLogout.setOnClickListener {
            prefs.edit().clear().apply()
            startActivity(Intent(this, LoginActivity::class.java))
            finish()
        }
    }
}
